development/prow-installer/pkg/kubectl/wrapper.go

```diff
@@ -0,0 +1,104 @@
+package kubectl
+
+import (
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+type Wrapper struct {
+	Kubeconfig string
+}
+
+func (c *Wrapper) Apply() *KubeApplyCommand {
+	return &KubeApplyCommand{Config: c.Kubeconfig}
+}
+
+type KubeApplyCommand struct {
+	Type   string
+	Path   string
+	Nspace string
+	Config string
+}
+
+func (kac *KubeApplyCommand) File(path string) *KubeApplyCommand {
+	kac.Type = "-f" //file
+	//change slashed path (dir/filename) to OS-dependent path.
+	kac.Path = filepath.FromSlash(path)
+	return kac
+}
+
+func (kac *KubeApplyCommand) Directory(path string) *KubeApplyCommand {
+	kac.Type = "-k" //directory
+	//change slashed path (dir/directory/) to OS-dependent path.
+	kac.Path = filepath.FromSlash(path)
+	return kac
+}
+
+func (kac *KubeApplyCommand) Do() error {
+	cmd := exec.Command("kubectl", "apply")
+	cmd.Env = append(os.Environ(), fmt.Sprintf("KUBECONFIG=%s", kac.Config))
+
+	if kac.Path == "" {
+		return fmt.Errorf("path cannot be empty")
+	}
+
+	cmd.Args = append(cmd.Args, kac.Type, kac.Path)
+
+	log.Debugf("executing command: %s", cmd.String())
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("error executing command: %w %s", err, out)
+	}
+	return nil
+}
+
+// generate kubeconfig based on credentials provided in arguments
+// the function returns path to the config file.
+// it's needed to have GOOGLE_CREDENTIALS_APPLICATION env variable set
+func GenerateKubeconfig(endpoint, cadata, name string) (string, error) {
+	if endpoint == "" {
+		return "", fmt.Errorf("endpoint cannot be empty")
+	}
+	if cadata == "" {
+		return "", fmt.Errorf("cadata cannot be empty")
+	}
+	if name == "" {
+		return "", fmt.Errorf("name cannot be empty")
+	}
+
+	path := filepath.FromSlash(".kube/" + name + "_config")
```

Comment:
For here I believe it's better to have the working directory of the application explicitly: https://golang.org/pkg/os/#Getwd

Reply:
I was thinking about it and indeed rooting the path should be better. Something for ex.:
```go
cwd, _ := os.Getwd()
path := filepath.FromSlash(cwd + ".kube/" + name + "_config")
```